File: tests/fixtures/swift/inheritance.swift
protocol Serializable {}
protocol Logging {
    func log(msg: String)
}
protocol Renderable {
    func render()
}

class Service: Serializable, Logging, Renderable {
    func log(msg: String) {
        print(msg)
    }

    func render() {}
}

class Base {
    var id: Int

    init(id: Int) {
        self.id = id
    }
}

class Child: Base, Logging {
    var extra: String

    init(id: Int, extra: String) {
        self.extra = extra
        super.init(id: id)
    }

    func log(msg: String) {
        print("[\(id)] \(msg)")
    }
}

File: tests/fixtures/scala/inheritance.scala
package com.example.inheritance

trait Serializable
trait Logging {
  def log(msg: String): Unit
}
trait Drawable {
  def draw(): Unit
}

class Service(val name: String) extends Serializable with Logging with Drawable {
  override def log(msg: String): Unit = println(msg)
  override def draw(): Unit = ()
}

abstract class Base(val id: Int)
class Child(id: Int, val extra: String) extends Base(id) with Logging {
  override def log(msg: String): Unit = println(s"[$id] $msg")
}
